### PR TITLE
Canvas: Fix button element size not being applied correctly

### LINKED ISSUE #74225: Canvas: Button size should scale with the size of the element
https://github.com/grafana/grafana/assets/22381771/8e07536c-9c87-47f5-a91f-f153770e50ed

### PR DESCRIPTION
**Pull request by linghaoSu**
**Which issue(s) does this PR fix?**:

set button size 100% to fill container dom

Fixes #74225

**Special notes for your reviewer:**

Please check that:
- [x] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

**Comment by ryantxu** [comment]
hymm -- maybe this should be optional, would not be great if all buttons need to be explicitly sized to get normal design system/theme based buttons 🤔

**Comment by linghaoSu** [comment]
> hymm -- maybe this should be optional, would not be great if all buttons need to be explicitly sized to get normal design system/theme based buttons 🤔

Maybe add a config that determines if the size of the button is controlled based on the size of the container?

**Review by nmarrs — APPROVED** [review]
This looks good to me - per comment around standardized button sizes I think could handle this in the future via a specific button element option around sizing 

Playing around with this locally it seems that the button text is always left aligned, I'll add a follow up PR that centers the text for the time being (we may also want to consider adding text alignment options but we can wait off on that / see if that is a requested feature)

**Comment by nmarrs** [comment]
@linghaoSu are you able to rebase this branch on to latest `main`?

`git merge origin main`

That will fix the build and allow us to merge your contribution :)

**Comment by linghaoSu** [comment]
> @linghaoSu are you able to rebase this branch on to latest `main`?
> 
> `git merge origin main`
> 
> That will fix the build and allow us to merge your contribution :)

Okay, I've done the merge from the main branch.

File: public/app/features/canvas/elements/button.tsx [hunk 1]
```diff
@@ -40,20 +43,30 @@ export const defaultStyleConfig: ButtonStyleConfig = {
...
   }
 }
 
+const getStyles = stylesFactory((theme: GrafanaTheme2, data: ButtonData | undefined) => ({
+  button: css`
+    position: absolute;
+    height: 100%;
+    width: 100%;
+  `,
+}));
```
Comment (nmarrs) [inline public/app/features/canvas/elements/button.tsx:68]: Build is failing due to a lint issue where we expect styles to written as objects 😅 after fixing this the build should go green 🤞 
```suggestion
const getStyles = stylesFactory((theme: GrafanaTheme2, data: ButtonData | undefined) => ({
  button: css({
    position: 'absolute',
    height: '100%',
    width: '100%',
  }),
}));
```

**Pull request merged**